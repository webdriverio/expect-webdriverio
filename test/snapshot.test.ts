import { test, expect } from 'vitest'
import type { Frameworks } from '@wdio/types'

import { expect as expectExport, SnapshotService } from '../src/index.js'

const service = SnapshotService.initiate()
service.beforeTest({
    title: 'test',
    parent: 'parent',
    file: '/foo/bar/file',
} as Frameworks.Test)

test('supports snapshot testing', () => {
    const exp = expectExport;
    expect(exp).toBeDefined()
    expect(exp({}).toMatchSnapshot).toBeDefined()
    expect(exp({}).toMatchInlineSnapshot).toBeDefined()
    exp({ a: 'a' }).toMatchSnapshot()
    exp({ a: 'a' }).toMatchInlineSnapshot()
})
