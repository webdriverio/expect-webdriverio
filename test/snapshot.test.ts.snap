// Snapshot v1

exports[`parent > test 1`] = `
{
  "a": "a",
}
`;
